services/QuillLMS/app/services/analytics/segment_analytics.rb

```diff
@@ -72,11 +72,18 @@ def track_activity_pack_assignment(teacher_id, unit_id)
   end
 
   def track_activity_completion(user, student_id, activity, activity_session)
-    track({
-      user_id: user&.id,
-      event: SegmentIo::BackgroundEvents::ACTIVITY_COMPLETION,
-      properties: activity.segment_activity.content_params.merge({student_id: student_id})
-    })
+    activity_topics = activity.topics
+    activity_topics.each do |topic|
+      level_one_topic = topic
+      level_two_topic = Topic.find(topic.parent_id)
```

Comment:
Some refactoring suggestions for consistency and also to break up the long line:

```
  activity_topics.each do |topic_level_one|
    topic_level_two = Topic.find(topic_level_one.parent_id)
    topic_level_three = Topic.find(topic_level_two.parent_id)

    properties = { 
      student_id: student_id, 
      topic_level_one: topic_level_one.name, 
      topic_level_two: topic_level_two.name, 
      topic_level_three: topic_level_three.name, 
    }

    track({
        user_id: user&.id,
        event: SegmentIo::BackgroundEvents::ACTIVITY_COMPLETION,
        properties: activity.segment_activity.content_params.merge(properties)
    })
  ```